extensions/EXT/EGL_EXT_surface_compression.txt

```diff
@@ -0,0 +1,289 @@
+
+   [Only if EGL_EXT_yuv_surface is supported]
+
+    If _config_ describes a surface with multiple planes (the
+    value of the EGL_YUV_NUMBER_OF_PLANES_EXT attribute is larger than one),
+    then the fixed-rate compression rate can be specified independently for
+    the each plane.
+    In this case, EGL_SURFACE_COMPRESSION_EXT specifies the fixed-rate
+    compression that may be enabled for rendering to plane 0,
+    EGL_SURFACE_COMPRESSION_PLANE1_EXT specifies the fixed-rate
+    compression that may be enabled for rendering to plane 1, and
+    EGL_SURFACE_COMPRESSION_PLANE2_EXT specifies the fixed-rate compression
+    that may be enabled for rendering to plane 2.
+    The supported values of EGL_SURFACE_COMPRESSION_PLANE1_EXT and
+    EGL_SURFACE_COMPRESSION_PLANE2_EXT are the same as for
+    EGL_SURFACE_COMPRESSION_EXT.
+
+    If _config_ has more than one plane and the
+    EGL_SURFACE_COMPRESSION_PLANE1_EXT attribute is not specified,
+    then the value of EGL_SURFACE_COMPRESSION_EXT is used for all planes.
+    If _config_ has more than two planes and the
+    EGL_SURFACE_COMPRESSION_PLANE2_EXT attribute is not specified,
+    then the value of EGL_SURFACE_COMPRESSION_PLANE1_EXT is also used
+    for plane 2.
+
+    The default value of EGL_SURFACE_COMPRESSION_PLANE1_EXT and
+    EGL_SURFACE_COMPRESSION_PLANE2_EXT is EGL_SURFACE_COMPRESSION_FIXED_RATE_NONE_EXT.
+
+    Modify section 3.5.6 "Surface Attributes"
+
+    Add entries to Table 3.5 "Queryable surface attributes and types":
+         
+        EGL_SURFACE_COMPRESSION_EXT            enum    Fixed-rate compression
+
+   [Only if EGL_EXT_yuv_surface is supported]
+        EGL_SURFACE_COMPRESSION_PLANE1_EXT     enum    Fixed-rate compression for plane 1
+        EGL_SURFACE_COMPRESSION_PLANE2_EXT     enum    Fixed-rate compression for plane 2
+
+    Add the following paragraph:
+
+    Querying EGL_SURFACE_COMPRESSION_EXT returns the actual fixed-rate
+    compression applied to a surface. For YUV surfaces, the value applied to
+    the luma plane is returned. This value may be different to the one
+    requested when the surface was created.
+    For pbuffer and pixmap surfaces, the contents of <value> are not modified."
+
+    [Only if EGL_EXT_yuv_surface is supported]
+    Querying EGL_SURFACE_COMPRESSION_PLANE1_EXT returns the actual
+    fixed-rate compression applied to plane 1 of a YUV surface.
+    Querying EGL_SURFACE_COMPRESSION_PLANE2_EXT returns the actual
+    fixed-rate compression applied to plane 2 of a YUV surface.
+    These values may be different to the one requested when the surface was created.
+    For pbuffer and pixmap surfaces, the contents of <value> are not modified."
+
+    To get the list of all fixed-rate compression rates that are available on
+    a specified display and EGLConfig, call
+
+       EGLBoolean eglQuerySupportedCompressionRatesEXT(
+           EGLDisplay dpy, EGLConfig config, const EGLAttrib *attrib_list,

```

Comment:
config is the format, rates are compression options on top of that? And window System can decode the compression from the buffer?

Reply:
Yes. (Some overlap with the reply to the previous question.)